### PR TITLE
[TM-140] View Application Details

### PR DESCRIPTION
**Pull request by mmiddleton3301**
Introduces the "ApplicationStatus" page, which pulls its data from its relevant APIM endpoint.

Note: currently targetting the TM-45 branch.

See also:

* https://github.com/SkillsFundingAgency/das-apim-endpoints/pull/497
* https://github.com/SkillsFundingAgency/das-levy-transfer-matching-api/pull/44

### CODE REVIEW

**Review by chrisfoster76 — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/ApplicationStatus.cshtml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+@model SFA.DAS.LevyTransferMatching.Web.Models.Applications.ApplicationStatusViewModel
+
+@{
+    ViewData["Title"] = $"Your {(Model.IsNamePublic ? Model.EmployerAccountName : "opportunity")} ({Model.EncodedOpportunityId}) application details";
+}
+
+<a href="@Url.Action("GetApplications")" class="govuk-back-link">Back</a>
+<main class="govuk-main-wrapper govuk-main-wrapper--auto-spacing" id="main-content">
+
+    <div class="govuk-width-container form-entry">
```
Comment (jamesmking) [inline src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/ApplicationStatus.cshtml:12]: Can we take out this `<div>` there is already a `govuk-width-container` in the layout file. When we have more than 1 of these, we get extra padding/margin on smaller sized screens

File: src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/ApplicationsService/ApplicationsService.cs [hunk 2]
```diff
@@ -22,5 +24,25 @@ public async Task<GetApplicationsResponse> GetApplications(long accountId, Cance
 
             return JsonConvert.DeserializeObject<GetApplicationsResponse>(await response.Content.ReadAsStringAsync());
         }
+
+        public async Task<GetApplicationStatusResponse> GetApplicationStatus(long accountId, int applicationId)
```
[the remark this answers is not in the document]
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/ApplicationsService/ApplicationsService.cs]: Yeah, now you point it out, it is an odd name. I've updated it.

File: src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/ApplicationStatus.cshtml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+﻿@using SFA.DAS.LevyTransferMatching.Domain.Extensions
+
+@model SFA.DAS.LevyTransferMatching.Web.Models.Applications.ApplicationStatusViewModel
+
+@{
+    ViewData["Title"] = $"Your {(Model.IsNamePublic ? Model.EmployerAccountName : "opportunity")} ({Model.EncodedOpportunityId}) application details";
```
[the remark this answers is not in the document]
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/ApplicationStatus.cshtml]: Done.

File: src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/ApplicationStatus.cshtml (hunk 1 above)
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/ApplicationStatus.cshtml]: Thanks @jamesmking - removed this extra div 👍

File: src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/ApplicationStatus.cshtml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                </div>
+
+                <div class="accepted-text-hide">
+                    <h2 class="govuk-heading-m">About your apprenticeship training</h2>
+                </div>
+
+                <section class="das-highlight">
+                    <p><strong>Apprenticeship training course:</strong> @Model.JobRole</p>
+                    <p><strong>Number of apprentices:</strong> @Model.NumberOfApprentices</p>
+                    <p><strong>Estimated total cost:</strong> @Model.Amount</p>
```
[the remark this answers is not in the document]
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/ApplicationStatus.cshtml]: This has been changed to the actual estimate.

File: src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/Application.cshtml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+﻿@using SFA.DAS.LevyTransferMatching.Domain.Extensions
+
+@model SFA.DAS.LevyTransferMatching.Web.Models.Applications.ApplicationViewModel
+
+@{
+    ViewData["Title"] = Model.Title;
+}
+
+<a href="@Url.Action("GetApplications")" class="govuk-back-link">Back</a>
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/Application.cshtml:9]: This back link doesn't work

**Review by chrisfoster76 — CHANGES_REQUESTED** [review]
One minor issue with back link

File: src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/Application.cshtml (hunk 5 above)
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/Application.cshtml]: Good spot 👀 Fixed

**Review by chrisfoster76 — APPROVED** [review]

**Pull request merged**